MyOnboarding/src/MyOnboardingApp.Services/Services/CreateItemService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading.Tasks;
+using MyOnboardingApp.Contracts.Generators;
+using MyOnboardingApp.Contracts.Models;
+using MyOnboardingApp.Contracts.Repository;
+using MyOnboardingApp.Contracts.Services;
+using MyOnboardingApp.Contracts.Validation;
+
+namespace MyOnboardingApp.Services.Services
+{
+    internal class CreateItemService : AbstractUpsertService, ICreateItemService
+    {
+        private readonly ITodoListRepository _repository;
+        private readonly IIdGenerator<Guid> _idGenerator;
+        private readonly IDateTimeGenerator _dateTimeGenerator;
+
+
+        public CreateItemService(ITodoListRepository repository, IIdGenerator<Guid> idGenerator, IDateTimeGenerator dateTimeGenerator, IInvariantValidator<TodoListItem> validator)
+            : base(validator)
+        {
+            _repository = repository;
+            _idGenerator = idGenerator;
+            _dateTimeGenerator = dateTimeGenerator;
+        }
+
+
+        public async Task<IItemWithErrors<TodoListItem>> AddNewItemAsync(TodoListItem newItem)
+        {
+            var initializingData = GetGeneratedData(newItem);
+            return await TryCompleteAndStoreItemAsync(newItem, initializingData);
+        }
+
+
+        protected override async Task StoreToDatabase(TodoListItem completedItem)
+            => await _repository.AddNewItemAsync(completedItem);
+
+
+        protected virtual ItemInitializingData GetGeneratedData(TodoListItem originalItem)
```

Comment:
Same as the [`UpdateItemService`](https://github.com/petrareb/kentico-onboarding-cs/pull/4/files#diff-509d50ba5b162c08314cf315832c621bR47) ("including the part" where you don't need the `TodoListItem` 😛)